lib/models/bucket.dart

```diff
@@ -7,55 +6,58 @@ import 'package:vikunja_app/models/user.dart';
 @JsonSerializable()
 class Bucket {
   int id, listId, limit;
-  String? title;
-  double position;
-  DateTime? created, updated;
-  User? createdBy;
+  String title;
+  double? position;
+  late final DateTime created, updated;
+  User createdBy;
   bool isDoneBucket;
+  late final List<Task> tasks;
+
   Bucket({
-    required this.id,
+    this.id = -1,
     required this.listId,
-    this.title,
-    this.position = 0,
+    required this.title,
+    this.position,
     required this.limit,
     this.isDoneBucket = false,
-    this.created,
-    this.updated,
-    this.createdBy,
-    this.tasks = const <Task>[],
-  });
-
-  List<Task> tasks = [];
+    DateTime? created,
+    DateTime? updated,
+    required this.createdBy,
+    List<Task>? tasks,
+  }) {
+    this.created = created ?? DateTime.now();
+    this.updated = created ?? DateTime.now();
+    this.tasks = tasks ?? [];
```

Comment:
`const` lists are not growable, where `final` lists are. `tasks` is currently manipulated directly when moving tasks between buckets, so it can't be `const`.